app/assets/javascripts/admin/modules/breadcrumb_preview.js

```diff
@@ -0,0 +1,66 @@
+(function(Modules) {
+  "use strict";
+
+  Modules.BreadcrumbPreview = function() {
+    var that = this;
+
+    that.fetchCurrentCheckboxes = function() {
+      return $('.topic-tree :checked');
+    };
+
+    that.buildBreadcrumbsStructure = function(checkboxes) {
+      var structure = $.map(checkboxes, function(checkbox) {
+        var $checkbox = $(checkbox);
+        var ancestors = $checkbox.data('ancestors').split('|');
+
+        return [ancestors];
+      });
+
+      return structure;
+    };
+
+    that.filterBreadcrumbs = function(breadcrumbs) {
+      var longestFirst = breadcrumbs.sort(function (a, b) {
+        return b.length - a.length;
+      });
+
+      var seenAlready = [];
+
+      $.each(longestFirst, function(index, breadcrumb) {
+        var seen = false;
+
+        $.each(seenAlready, function(index, seenBreadcrumb) {
+          var breadcrumbString = JSON.stringify(breadcrumb);
```

Comment:
You can't compare two arrays in Javascript, using `JSON.stringify` seemed to be the only reasonable solution, fun facts on the screenshot underneath
![screen shot 2017-02-17 at 15 06 14](https://cloud.githubusercontent.com/assets/136777/23074649/6562d274-f531-11e6-8a84-7cb88c6fcbb4.png)